File: ios/Classes/Codec/DrawOnImageReader.swift
class DrawOnImageReader : FlutterStandardReader {
    override func readValue(ofType type: UInt8) -> Any? {
        switch type {
        case drawOnImageTypeId:
            let stringToWrite = super.readValue() as! String
            let left = super.readValue() as! NSNumber
            let right = super.readValue() as! NSNumber
            let top = super.readValue() as! NSNumber
            let bottom = super.readValue() as! NSNumber
            let color = super.readValue() as! NSNumber
            let size = super.readValue() as! NSNumber
            let imageCanvas = super.readValue() as! FlutterStandardTypedData
            return DrawOnImageData(
                stringToWrite: stringToWrite,
                leftPadding: left.intValue,
                rightPadding: right.intValue,
                topPadding: top.intValue,
                bottomPadding: bottom.intValue,
                fontColor: color.intValue,
                fontSize: size.intValue,
                imageCanvas: imageCanvas.data
            )
        default:
            return super.readValue(ofType: type)
        }
    }
}

File: android/src/main/kotlin/com/nikolai/imagedraw/draw_on_image_plugin/codec/DrawOnImageReader.kt
package com.nikolai.imagedraw.draw_on_image_plugin.codec

import com.nikolai.imagedraw.draw_on_image_plugin.model.DrawOnImageData
import com.nikolai.imagedraw.draw_on_image_plugin.utils.DRAW_ON_IMAGE_DATA_TYPE
import io.flutter.plugin.common.StandardMessageCodec
import java.nio.ByteBuffer

class DrawOnImageReader : StandardMessageCodec() {
    public override fun readValueOfType(type: Byte, buffer: ByteBuffer?): Any {
        println("It's type : $type")
        return when (type) {
            DRAW_ON_IMAGE_DATA_TYPE -> {
                val text = super.readValue(buffer) as String
                val left = super.readValue(buffer) as Int
                val right = super.readValue(buffer) as Int
                val top = super.readValue(buffer) as Int
                val bottom = super.readValue(buffer) as Int
                val color = super.readValue(buffer) as Long
                val size = super.readValue(buffer) as Int
                val imageCanvas = super.readValue(buffer) as ByteArray

                DrawOnImageData(
                        text,
                        left,
                        right,
                        top,
                        bottom,
                        color.toInt(),
                        size,
                        imageCanvas
                )
            }
            else -> super.readValueOfType(type, buffer)
        }
    }
}